test/reader.test.js

```diff
@@ -33,175 +23,139 @@ function getExecutionOrder (bundle) {
-        const feedB = sink.reader('simple.b.json');
-        const missingFile = createErrorStream(sink, 'invalid-file.json', 200);
-
-        const reader = new Reader([feedA, feedB, missingFile]);
-        return reader;
-    }
-
-    const [bundle] = await Promise.all([getStream(createReader())]);
-
-    const executionOrder = getExecutionOrder(bundle);
-
-    t.deepEqual(executionOrder, [
-        FEED_A_HASH,
-        FEED_B_HASH,
+test('code reach 3 entry point', async () => {
+    const result = await bundleJS([
+        [
+            {
+                entry: true,
+                id: 'a',
+                source: 'spy("a");',
+            },
+        ], [
+            {
+                entry: true,
+                id: 'b',
+                source: 'spy("b");',
+            },
+        ], [
+            {
+                entry: true,
+                id: 'c',
+                source: 'spy("c");',
+            },
+        ],
     ]);
-    t.snapshot(prettier.format(bundle));
+    const spy = jest.fn();
+    vm.runInNewContext(result, { spy });
+    expect(spy).toHaveBeenCalledTimes(3);
+    expect(spy.mock.calls).toMatchSnapshot();
 });
 
+test('should exclude/dedupe common modules', async () => {
+    const result = await bundleJS([
+        [
+            {
+                entry: true,
+                id: 'a',
+                source: 'require("./c");spy("a");',
+                deps: { './c': 'c' },
+            },
+            {
+                id: 'c',
+                source: 'spy("c");',
+                deps: {},
+            },
+        ], [
+            {
+                entry: true,
+                id: 'b',
+                source: 'spy("b");',
+            },
+            {
+                id: 'c',
+                source: 'require("./c");spy("c");',
+                deps: { './c': 'c' },
+            },
+        ],
+    ]);
+    const spy = jest.fn();
+    vm.runInNewContext(result, { spy });
 
-test('should keep ordering if very slow', async (t) => {
-    const { sink } = t.context;
-
-    const feedA = createSlowStream(sink, 'simple.a.json', 500);
-    const feedB = sink.reader('simple.b.json');
-    const feedC = createSlowStream(sink, 'simple.c.json', 100);
-
-    const reader = new Reader([feedA, feedB, feedC]);
-
-    const pipeline = new Promise((resolve, reject) => {
-        reader.on('pipeline empty', reject);
-        reader.on('pipeline ready', resolve);
-    });
-
-    const [bundle] = await Promise.all([getStream(reader), pipeline]);
-
-    const executionOrder = getExecutionOrder(bundle);
+    expect(spy).toHaveBeenCalledTimes(3);
+    expect(spy.mock.calls).toMatchSnapshot();
+    expect(prettier.format(result)).toMatchSnapshot();
+});
 
-    t.deepEqual(executionOrder, [
-        FEED_A_HASH,
-        FEED_B_HASH,
-        FEED_C_HASH,
-    ]);
-    t.snapshot(prettier.format(bundle));
+test('should error if no feed content', async () => {
+    expect(bundleJS()).rejects.toMatchSnapshot();
```

Comment:
you have to `await` when using `rejects`. And I think you need to do try-catch for now, unfortunately.

https://github.com/facebook/jest/pull/4962

Reply:
See the snapshots, it worked I think.

Reply:
```
exports[`should error if no feed content 1`] = `[AssertionError [ERR_ASSERTION]: Expected an array of feed arrays, got "undefined"]`;

exports[`should error if no feed content 2`] = `[AssertionError [ERR_ASSERTION]: Expected at least 1 feed]`;

exports[`should error if no feed content 3`] = `[AssertionError [ERR_ASSERTION]: Expected at least 1 feed with more than one entry]`;

exports[`should error if no feed content 4`] = `[AssertionError [ERR_ASSERTION]: Expected feed entry to have an id, got "undefined"]`;
```